spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/impl/ClassFactory.java

```diff
@@ -107,20 +106,22 @@ static IScannableCodeBase createFilesystemCodeBase(FilesystemCodeBaseLocator cod
 
         File file = new File(fileName);
 
-        if (!file.exists()) {
-            throw new FileNotFoundException("File " + file.getAbsolutePath() + " doesn't exist");
-        } else if (!file.canRead()) {
-            throw new IOException("File " + file.getAbsolutePath() + " not readable");
+        if (!file.exists() || !file.canRead()) {
+            return new EmptyCodeBase(codeBaseLocator);
         } else if (file.isDirectory()) {
             return new DirectoryCodeBase(codeBaseLocator, file);
         } else if (!file.isFile()) {
-            throw new IOException("File " + file.getAbsolutePath() + " is not a normal file");
+            return new EmptyCodeBase(codeBaseLocator);
```

Comment:
I think we need to log the absolute path of `file`, to help debugging.

Reply:
Thanks for the review! I can definitely add some logging. I am not sure of the best way to add the logging. It seems that the `java.util.Logger` is used infrequently in the code and this doesn't fall under the use case of `IErrorLogger`.

Any recommendations for the best logging mechanism in this situation?

Reply:
Hi @KengoTODA. Trying a ping to see if you have thoughts on how best to log in this situation. Thanks!

Reply:
Now SpotBugs uses `System.out` to log, you can refer [FindBugs2.java](https://github.com/spotbugs/spotbugs/blob/release-3.1/spotbugs/src/main/java/edu/umd/cs/findbugs/FindBugs2.java#L906-L908) for example.

note: When we log error, we can take `IErrorLogger` from `AnalysisContext` or `AnalysisCache`, but it doesn't support logging warning.

Reply:
If we're sure that this log is needless for normal use, we can log it only when `FindBugs2.DEBUG` is true.

Reply:
Change made to log to `System.out` when `Findbugs2.DEBUG` is enabled.